src/Microsoft.Health.Fhir.Core/Features/Search/Registry/SearchParameterStatusManager.cs

```diff
@@ -88,63 +88,43 @@ public async Task EnsureInitialized()
-                            }
-                        }
-                    }
+            await _mediator.Publish(new SearchParametersUpdated(updated));
+        }
 
-                    if (p.BaseResourceTypes != null)
-                    {
-                        foreach (string resourceType in p.BaseResourceTypes)
-                        {
-                            if (resourceTypeSearchParamStatusMap.ContainsKey(resourceType))
-                            {
-                                resourceTypeSearchParamStatusMap[resourceType].Add(result);
-                            }
-                            else
-                            {
-                                resourceTypeSearchParamStatusMap.Add(resourceType, new List<ResourceSearchParameterStatus>() { result });
-                            }
-                        }
-                    }
-                }
-            }
+        public async Task UpdateSearchParameterStatus(IReadOnlyCollection<string> searchParameterUris, SearchParameterStatus status)
+        {
+            var searchParameterStatusList = new List<ResourceSearchParameterStatus>();
+            var updated = new List<SearchParameterInfo>();
 
-            var resourceHashMap = new Dictionary<string, string>();
-            foreach (KeyValuePair<string, List<ResourceSearchParameterStatus>> kvp in resourceTypeSearchParamStatusMap)
+            foreach (string uri in searchParameterUris)
             {
-                string searchParamHash = SearchHelperUtilities.CalculateSearchParameterHash(kvp.Value);
-                resourceHashMap.Add(kvp.Key, searchParamHash);
+                var searchParamUri = new Uri(uri);
+
+                var paramInfo = _searchParameterDefinitionManager.GetSearchParameter(searchParamUri);
+                updated.Add(paramInfo);
+                paramInfo.IsSearchable = status == SearchParameterStatus.Enabled;
+                paramInfo.IsSupported = status != SearchParameterStatus.Disabled;
+
+                searchParameterStatusList.Add(new ResourceSearchParameterStatus()
+                {
+                    LastUpdated = Clock.UtcNow,
+                    Status = status,
+                    Uri = searchParamUri,
+                });
             }
 
-            await _mediator.Publish(new SearchParametersHashUpdated(resourceHashMap));
+            await _searchParameterStatusDataStore.UpsertStatuses(searchParameterStatusList);
 
             await _mediator.Publish(new SearchParametersUpdated(updated));
         }
+
+        public async Task AddSearchParameterStatus(string searchParamUri)
+        {
+            // new search parameters are added as supported, until reindexing occurs, when
+            // they will be fully enabled
+            await UpdateSearchParameterStatus(new List<string>() { searchParamUri }, SearchParameterStatus.Supported);
```

Comment:
Silly question: would there ever be a time that a user would want to add a search parameter that is not supported?

Reply:
I think there could be a case when you might want to.  Perhaps you are adding a profile or implementation guide, such as US Core, and it just happens that we don't support one of the parameters.  You may not want the entire profile to be rejected because of one unsupported parameter, and perhaps you want the parameter stored for reference, even if you can't search on it.  However, for the moment, to get to an MVP of custom search parameters, we are going to reject any unsupported parameters.

Reply:
Makes sense, thanks for the info!